src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -101,7 +105,56 @@ export const CriteriaView = ({ criteria, deleteCriteria, updateCriteria }: Crite
                   }),
                 ]);
               case 'range':
-                return h(Fragment, [strong([`${criteria.name}:`]), ` ${criteria.low} - ${criteria.high}`]);
+                const numberInputStyles = {
+                  width: '4rem',
+                  padding: 0,
+                };
+                const rangeSliderMargin = 20;
+                return h(Fragment, [
+                  div([strong([`${criteria.name}:`]), ` ${criteria.low} - ${criteria.high}`]),
+                  div({ style: { display: 'flex', alignItems: 'center' } }, [
+                    h(NumberInput, {
+                      min: criteria.rangeOption.min,
+                      max: criteria.high,
+                      isClearable: false,
+                      onlyInteger: true,
+                      value: criteria.low,
+                      onChange: (v) => updateCriteria({ ...criteria, low: v }),
+                      'aria-label': `${criteria.name} low`,
+                      style: numberInputStyles,
+                    }),
+                    h(Slider, {
+                      range: true,
+                      value: [criteria.low, criteria.high],
+                      min: criteria.rangeOption.min,
+                      onChange: (values) => updateCriteria({ ...criteria, low: values[0], high: values[1] }),
+                      max: criteria.rangeOption.max,
+                      style: { marginLeft: rangeSliderMargin },
+                      handleStyle: {
+                        borderColor: colors.accent(),
```

Comment:
I think this would be a little easier to read if these styles were put into a const like `flexWithBaseline`, but that's just my preference!